### ISSUE TITLE
tensorflow lite converter: why empty lite file

### ISSUE THREAD

**Issue by guoxiaolu** [tensorflow/tensorflow#15537]
I have a tensorflow pb file and try to use tflite converter to convert it to lite file,but the result file is empty and no error. I don't know why
`bazel run --config=opt --copt=-msse4.1 --copt=-msse4.2  //tensorflow/contrib/lite/toco:toco --   --input_file=/Users/Lavector/code/keras_to_tensorflow/mobilenet_regression.pb   --output_file=/tmp/mobilenet_regression.lite   --input_format=TENSORFLOW_GRAPHDEF   --output_format=TFLITE   --inference_type=FLOAT   --input_shape=1,224,224,3   --input_array=input   --output_array=output_node0`

`WARNING: Config values are not defined in any .rc file: opt
INFO: Analysed target //tensorflow/contrib/lite/toco:toco (0 packages loaded).
INFO: Found 1 target...
Target //tensorflow/contrib/lite/toco:toco up-to-date:
  bazel-bin/tensorflow/contrib/lite/toco/toco
INFO: Elapsed time: 0.223s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action

INFO: Running command line: bazel-bin/tensorflow/contrib/lite/toco/toco '--input_file=/Users/Lavector/code/keras_to_tensorflow/mobilenet_regression.pb' '--output_file=/tmp/mobilenet_regression.lite' '--input_format=TENSORFLOW_GRAPHDEF' '--output_format=TFLITE' '--inference_type=FLOAT' '--input_shape=1,224,224,3' '--input_array=input' '--output_array=output_node0'
2017-12-21 11:20:11.351544: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 472 operators, 695 arrays (0 quantized)
2017-12-21 11:20:11.384604: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After general graph transformations pass 1: 85 operators, 200 arrays (0 quantized)
2017-12-21 11:20:11.387067: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before dequantization graph transformations: 85 operators, 200 arrays (0 quantized)`

I use mobilenet to do regression, and it is successful when test.
I test the example code of [https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/cmdline_examples.md](url), and it is correct.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by guoxiaolu** [comment]
@tensorflowbutler 
It works well when use the tensorflow/android example, but not correct when tensorflow lite.

1. Have I written custom code: yes, I used keras mobile-net (it uses custom layer). Then I converted it to freezed graph. Finally, tried to use it by tf-lite.
2. OS Platform and Distribution: OS 10.11.6
3. TensorFlow installed from: git clone latest version
4. TensorFlow version: 1.4.0
5. Bazel version: 0.9.0-homebrew
I tested in cpu-only machine.
6. Exact command to reproduce: I just build "toco" and not build "freeze_graph", I also test the tf official mobilenet freezed graph by bazel-bin/tensorflow/contrib/lite/toco/toco, it was also not correct. so I finally used "bazel run" and the tf official mobilenet freezed graph generated correct lite file and work well in lite example, but my model generated empty file with no error.

I have also noticed some instruction of custom code, I don't know whether it is related.
If needed, the "pb" model can be supported for test.
Wish it is helpful. Thank you~

**Comment by GodsDusk** [comment]
I have same problem when use tensorflow.contrib.slim.python.slim.nets.resnet_v2 model 
```
bazel run --config=opt \
tensorflow/contrib/lite/toco:toco -- \
--input_file=/Users/ir/WORK/models/models/models/frozen_graph.pb \
--input_format=TENSORFLOW_GRAPHDEF \
--output_format=TFLITE \
--output_file=/Users/ir/WORK/models/models/models/nsfw.lite \
--inference_type=QUANTIZED_UINT8 \
--input_type=FLOAT \
--input_arrays=input \
--output_arrays=resnet_v2_50/predictions/Reshape_1 \
--input_shapes=1,224,224,3
```
2017-12-25 10:21:40.195960: W tensorflow/contrib/lite/toco/toco_cmdline_flags.cc:178] --input_type is deprecated. It was an ambiguous flag that set both --input_data_types and --inference_input_type. If you are trying to complement the input file with information about the type of input arrays, use --input_data_type. If you are trying to control the quantization/dequantization of real-numbers input arrays in the output file, use --inference_input_type.
2017-12-25 10:21:40.821969: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 433 operators, 634 arrays (0 quantized)


**OS Platform and Distribution:OS 10.13.1
TensorFlow installed from: git clone latest version
TensorFlow version: 1.4.1
Bazel version:0.9.0 homebrew
Python 2.7.14 :: Anaconda, Inc.**

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by andrehentz** [comment]
@guoxiaolu Would you be able to provide your .pb file?

**Comment by guoxiaolu** [comment]
https://pan.baidu.com/s/1jJDgfJK
This is my pb file

**Comment by andrehentz** [comment]
This looks like a bug that is triggered by specifying a nonexistent tensor as input. Perhaps you meant to specify "input_1" as your input array, instead of simply "input"?

**Comment by guoxiaolu** [comment]
It works, Thank you very much~

**Comment by Madhusakth** [comment]
Hi, 
I faced a similar error while building a tflite file from .pb file for speech model. I noticed that when I give --allow_custom_ops, it creates a tflite file of 6MB. But, without that, the created tflite files is zero bytes. 
When I upload this tflite filr with --allow_custom_ops, it gives a run time error asking for the custom ops to be defined. Could you please help me out with this? 

Thanks!

**Comment by guoxiaolu** [comment]
@Madhusakth , I think it's better to ask in a new issue and your problem is different from what I faced~ Good luck

**Comment by kingmacth** [comment]
@guoxiaolu 你是怎么解决的？我也同样的问题！
